src/kademlia/dht_storage.cpp

```diff
@@ -346,6 +345,7 @@ namespace
 		}
 
 		void put_immutable_item(sha1_hash const& target
+			, sha1_hash const& our_id
```

Comment:
There are a few ways of handling the fact that a node's id can change at any time:

1. Keep a copy of the node ids in dht_storage and provide a function to update them when an id changes. This creates duplicate storage which is undesirable both because it wastes memory and it runs the risk of the storage's id list getting out of sync with the actual ids.
2. Keep a reference to a data structure (likely a `std::vector<sha1_hash>`) which the owner of the storage instance is responsible for keeping up to date. Since we don't maintain such a list already, this has all the same problems as option 1.
3. Keep a reference to a function which has the necessary knowledge to reach into the existing data structures where the node id's are stored (i.e. `dht::node`). This is the best approach because it doesn't require duplicate storage and maintains loose coupling between the storage and dht node implementations.